packages/design-system/src/components/OcApplicationIcon/OcApplicationIcon.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div class="oc-application-icon oc-flex-inline oc-flex-middle oc-flex-center" :style="iconStyle">
+    <oc-icon :name="icon" :color="iconColor" size="medium" />
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  generateHashedColorForString,
+  getHexFromCssVar,
+  hexToRgb,
+  rgbToHex,
+  setDesiredContrastRatio,
+  shadeColor
+} from '../../helpers'
+import { computed, defineComponent, ref, unref } from 'vue'
+import OcIcon from '../OcIcon/OcIcon.vue'
+
+export default defineComponent({
+  components: { OcIcon },
+  name: 'OcApplicationIcon',
+  status: 'ready',
+  release: '15.0.0',
+  props: {
+    icon: {
+      type: String,
+      required: true
+    },
+    colorPrimary: {
+      type: String,
+      required: false,
+      default: ''
+    },
+    colorSecondary: {
+      type: String,
+      required: false,
+      default: ''
+    }
+  },
+
+  setup(props) {
+    const updateComputed = ref(false)
+    const iconColor = computed(() => {
+      return 'rgba(255,255,255,0.7)'
+    })
+    const getGradient = (primary: string, secondary: string): string => {
+      return `linear-gradient(90deg, ${primary} 0%, ${secondary} 100%)`
+    }
+    const primaryColor = computed(() => {
+      unref(updateComputed)
```

Comment:
is needed for reactivity, because the OcDrop doesn't destroy and rebuild the applicationMenu (like an v-if would), so when there is a theme switch the primaryColor stays the same

Reply:
Hmm that looks fishy. Also, I don't understand it 😄 `updateComputed` is being updated in `update()`, but this method is never beding called actually?

Edit: ahhhh now I see it, you're calling that method from the outside. Please avoid that 🙈  You could set a key on `oc-application-icon` which updates as soon as the drop opens. That way the component should be re-rendered.